### ISSUE TITLE
RedisMovedException when calling RKeysAsync#countExistsAsync

### ISSUE THREAD

**Issue by tfn-smartthings** [redisson/redisson#4395]
We have a scheduled task that runs (e.g. every 30 seconds) that collects some information about about a Redis stream. As part of this, we first check to see if the stream exists before proceeding. This check is done using `RKeysAsync#countExistsAsync`. When this is attempted it fails with the following error:
```
java.util.concurrent.CompletionException: org.redisson.client.RedisMovedException
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:332)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:347)
	at java.base/java.util.concurrent.CompletableFuture$BiRelay.tryFire(CompletableFuture.java:1498)
	at java.base/java.util.concurrent.CompletableFuture$CoCompletion.tryFire(CompletableFuture.java:1219)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2162)
	at org.redisson.command.RedisExecutor.handleError(RedisExecutor.java:508)
	at org.redisson.command.RedisExecutor.handleResult(RedisExecutor.java:490)
	at org.redisson.command.RedisExecutor.checkAttemptPromise(RedisExecutor.java:478)
	at org.redisson.command.RedisExecutor.lambda$execute$3(RedisExecutor.java:172)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2162)
	at org.redisson.client.protocol.CommandData.tryFailure(CommandData.java:87)
	at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:341)
	at org.redisson.client.handler.CommandDecoder.decodeCommand(CommandDecoder.java:198)
	at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:137)
	at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:113)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:510)
	at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:366)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:279)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1372)
	at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1235)
	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1284)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:510)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:449)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:279)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:722)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:658)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:584)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:496)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:995)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.redisson.client.RedisMovedException: null
	... 32 common frames omitted
```

Enabling debug logging for Redisson we see the additional lines (IPs hidden):
```
timestamp=2022-07-06T03:07:23,509Z, level=DEBUG, thread=in-msg-prism-stream-worker-1, logger=org.redisson.command.RedisExecutor, message=acquired connection for command (EXISTS) and params [in-msg-prism] from slot NodeSource [slot=null, addr=null, redisClient=null, redirect=null, entry=MasterSlaveEntry [masterEntry=[freeSubscribeConnectionsAmount=1, freeSubscribeConnectionsCounter=value:50:queue:0, freeConnectionsAmount=24, freeConnectionsCounter=value:64:queue:0, freezeReason=null, client=[addr=rediss://$IP_ADDR_A:6379], nodeType=MASTER, firstFail=0]]] using node $IP_ADDR_C/$IP_ADDR_C:6379... RedisConnection@966475896 [redisClient=[addr=rediss://$IP_ADDR_C:6379], channel=[id: 0xbc3845af, L:/$IP_ADDR_E:54634 - R:$IP_ADDR_C/$IP_ADDR_C:6379], currentCommand=null, usage=1]
timestamp=2022-07-06T03:07:23,509Z, level=DEBUG, thread=in-msg-prism-stream-worker-1, logger=org.redisson.command.RedisExecutor, message=acquired connection for command (EXISTS) and params [in-msg-prism] from slot NodeSource [slot=null, addr=null, redisClient=null, redirect=null, entry=MasterSlaveEntry [masterEntry=[freeSubscribeConnectionsAmount=1, freeSubscribeConnectionsCounter=value:50:queue:0, freeConnectionsAmount=24, freeConnectionsCounter=value:64:queue:0, freezeReason=null, client=[addr=rediss://$IP_ADDR_B:6379], nodeType=MASTER, firstFail=0]]] using node $IP_ADDR_D/$IP_ADDR_D:6379... RedisConnection@525234134 [redisClient=[addr=rediss://$IP_ADDR_D:6379], channel=[id: 0x9ef7b023, L:/$IP_ADDR_E:48986 - R:$IP_ADDR_D/$IP_ADDR_D:6379], currentCommand=null, usage=1]
timestamp=2022-07-06T03:07:23,514Z, level=DEBUG, thread=redisson-netty-2-16, logger=org.redisson.command.RedisExecutor, message=connection released for command (EXISTS) and params [in-msg-prism] from slot NodeSource [slot=null, addr=null, redisClient=null, redirect=null, entry=MasterSlaveEntry [masterEntry=[freeSubscribeConnectionsAmount=1, freeSubscribeConnectionsCounter=value:50:queue:0, freeConnectionsAmount=24, freeConnectionsCounter=value:64:queue:0, freezeReason=null, client=[addr=rediss://$IP_ADDR_A:6379], nodeType=MASTER, firstFail=0]]] using connection RedisConnection@966475896 [redisClient=[addr=rediss://$IP_ADDR_C:6379], channel=[id: 0xbc3845af, L:/$IP_ADDR_E:54634 - R:$IP_ADDR_C/$IP_ADDR_C:6379], currentCommand=CommandData [promise=java.util.concurrent.CompletableFuture@6ffb906c[Completed normally], command=(EXISTS), params=[in-msg-prism], codec=org.redisson.codec.MarshallingCodec], usage=0]
timestamp=2022-07-06T03:07:23,515Z, level=DEBUG, thread=redisson-netty-2-22, logger=org.redisson.command.RedisExecutor, message=connection released for command (EXISTS) and params [in-msg-prism] from slot NodeSource [slot=null, addr=null, redisClient=null, redirect=null, entry=MasterSlaveEntry [masterEntry=[freeSubscribeConnectionsAmount=1, freeSubscribeConnectionsCounter=value:50:queue:0, freeConnectionsAmount=24, freeConnectionsCounter=value:64:queue:0, freezeReason=null, client=[addr=rediss://$IP_ADDR_B:6379], nodeType=MASTER, firstFail=0]]] using connection RedisConnection@525234134 [redisClient=[addr=rediss://$IP_ADDR_D:6379], channel=[id: 0x9ef7b023, L:/$IP_ADDR_E:48986 - R:$IP_ADDR_D/$IP_ADDR_D:6379], currentCommand=CommandData [promise=java.util.concurrent.CompletableFuture@7eeb4e8[Completed exceptionally: org.redisson.client.RedisMovedException], command=(EXISTS), params=[in-msg-prism], codec=org.redisson.codec.MarshallingCodec], usage=0]
```

We are using AWS MemoryDB with the following configuration:
* 2 shards
* 2 nodes per shard (1 primary, 1 replica)

Application side, we are using the following configuration:
* cluster mode
* node address: AWS cluster endpoint -> 'clustercfg...amazonaws.com:6379'
* pingConnectionInterval: 30000
* timeout: 1500
* retryAttempts: 3
* retryInterval: 250
* tcpKeepAlive: true
* defaults for all other settings

We are using Redisson v3.17.4.

Can you advise possible solutions for this error? Is it possible we are using an incompatible configuration for our Redis setup?

Thanks for any help!

**Comment by mrniko** [comment]
Could you share logs in `trace` logging level?

**Comment by tfn-smartthings** [comment]
Here are some logs after enabling trace. I'm not able to copy everything since there is so much, but I tried to pull out Redisson logs that seemed related. Please let me know if there is a log in particular you would like that wasn't included in this snippet.

```
timestamp=2022-07-06T12:01:34,293Z, level=TRACE, thread=redisson-netty-2-32, logger=o.r.client.handler.CommandEncoder, message=channel: [id: 0xa1c6d1c6, L:/$IP_ADDR_A:35866 - R:$IP_ADDR_B/$IP_ADDR_B:6379] message: *1
$4
PING

timestamp=2022-07-06T12:01:34,294Z, level=TRACE, thread=redisson-netty-2-31, logger=o.r.client.handler.CommandEncoder, message=channel: [id: 0xce832432, L:/$IP_ADDR_A:58442 - R:$IP_ADDR_C/$IP_ADDR_C:6379] message: *1
$4
PING

timestamp=2022-07-06T12:01:34,295Z, level=TRACE, thread=redisson-netty-2-18, logger=o.r.client.handler.CommandEncoder, message=channel: [id: 0x7fe03d26, L:/$IP_ADDR_A:58448 - R:$IP_ADDR_C/$IP_ADDR_C:6379] message: *1
$4
PING

timestamp=2022-07-06T12:01:34,295Z, level=TRACE, thread=redisson-netty-2-16, logger=o.r.client.handler.CommandEncoder, message=channel: [id: 0x8b9b5414, L:/$IP_ADDR_A:35868 - R:$IP_ADDR_B/$IP_ADDR_B:6379] message: *1
$4
PING

timestamp=2022-07-06T12:01:34,295Z, level=TRACE, thread=redisson-netty-2-32, logger=o.r.client.handler.CommandDecoder, message=reply: +PONG
, channel: [id: 0xa1c6d1c6, L:/$IP_ADDR_A:35866 - R:$IP_ADDR_B/$IP_ADDR_B:6379], command: CommandData [promise=java.util.concurrent.CompletableFuture@4066610c[Not completed, 1 dependents], command=(PING), params=[], codec=org.redisson.client.codec.StringCodec]
timestamp=2022-07-06T12:01:34,296Z, level=TRACE, thread=redisson-netty-2-31, logger=o.r.client.handler.CommandDecoder, message=reply: +PONG
, channel: [id: 0xce832432, L:/$IP_ADDR_A:58442 - R:$IP_ADDR_C/$IP_ADDR_C:6379], command: CommandData [promise=java.util.concurrent.CompletableFuture@2d269cc7[Not completed, 1 dependents], command=(PING), params=[], codec=org.redisson.client.codec.StringCodec]
timestamp=2022-07-06T12:01:34,302Z, level=TRACE, thread=redisson-netty-2-16, logger=o.r.client.handler.CommandDecoder, message=reply: +PONG
, channel: [id: 0x8b9b5414, L:/$IP_ADDR_A:35868 - R:$IP_ADDR_B/$IP_ADDR_B:6379], command: CommandData [promise=java.util.concurrent.CompletableFuture@61ef061b[Not completed, 1 dependents], command=(PING), params=[], codec=org.redisson.client.codec.StringCodec]
timestamp=2022-07-06T12:01:34,303Z, level=TRACE, thread=redisson-netty-2-18, logger=o.r.client.handler.CommandDecoder, message=reply: +PONG
, channel: [id: 0x7fe03d26, L:/$IP_ADDR_A:58448 - R:$IP_ADDR_C/$IP_ADDR_C:6379], command: CommandData [promise=java.util.concurrent.CompletableFuture@2f378c1a[Not completed, 1 dependents], command=(PING), params=[], codec=org.redisson.client.codec.StringCodec]

...

timestamp=2022-07-06T12:01:34,512Z, level=TRACE, thread=redisson-netty-2-10, logger=o.r.client.handler.CommandEncoder, message=channel: [id: 0x9b8ac11d, L:/$IP_ADDR_A:38902 - R:$IP_ADDR_D/$IP_ADDR_D:6379] message: *2
$7
CLUSTER
$5
NODES

timestamp=2022-07-06T12:01:34,513Z, level=TRACE, thread=redisson-netty-2-10, logger=o.r.client.handler.CommandDecoder, message=reply: $809
38f34fe6872e971d793f836de6d0e4255e416692 cluster-001-0002-001.prt2ri.memorydb.us-east-1.amazonaws.com:6379@1122 slave 62c1c049cc51267b00ef8850d6300b254f156038 0 1657108892000 3 connected
62c1c049cc51267b00ef8850d6300b254f156038 cluster-001-0002-002.prt2ri.memorydb.us-east-1.amazonaws.com:6379@1122 master - 0 1657108892887 3 connected 8192-16383
7140d504912dbc76792e8e5c489aee0ee4db9346 cluster-001-0001-001.prt2ri.memorydb.us-east-1.amazonaws.com:6379@1122 slave 0e9a33460077513745c4dcf26e4756e041a2e083 0 1657108893847 4 connected
0e9a33460077513745c4dcf26e4756e041a2e083 cluster-001-0001-002.prt2ri.memorydb.us-east-1.amazonaws.com:6379@1122 myself,master - 0 1657108893000 4 connected 0-8191

...

timestamp=2022-07-06T12:01:39,069Z, level=DEBUG, thread=in-msg-prism-stream-worker-2, logger=org.redisson.command.RedisExecutor, message=acquired connection for command (EXISTS) and params [in-msg-prism] from slot NodeSource [slot=null, addr=null, redisClient=null, redirect=null, entry=MasterSlaveEntry [masterEntry=[freeSubscribeConnectionsAmount=1, freeSubscribeConnectionsCounter=value:50:queue:0, freeConnectionsAmount=24, freeConnectionsCounter=value:64:queue:0, freezeReason=null, client=[addr=rediss://$IP_ADDR_E:6379], nodeType=MASTER, firstFail=0]]] using node $IP_ADDR_B/$IP_ADDR_B:6379... RedisConnection@1735807760 [redisClient=[addr=rediss://$IP_ADDR_B:6379], channel=[id: 0xaa7da4bb, L:/$IP_ADDR_A:35790 - R:$IP_ADDR_B/$IP_ADDR_B:6379], currentCommand=null, usage=1]
timestamp=2022-07-06T12:01:39,069Z, level=DEBUG, thread=in-msg-prism-stream-worker-2, logger=org.redisson.command.RedisExecutor, message=acquired connection for command (EXISTS) and params [in-msg-prism] from slot NodeSource [slot=null, addr=null, redisClient=null, redirect=null, entry=MasterSlaveEntry [masterEntry=[freeSubscribeConnectionsAmount=1, freeSubscribeConnectionsCounter=value:50:queue:0, freeConnectionsAmount=24, freeConnectionsCounter=value:64:queue:0, freezeReason=null, client=[addr=rediss://$IP_ADDR_D:6379], nodeType=MASTER, firstFail=0]]] using node $IP_ADDR_C/$IP_ADDR_C:6379... RedisConnection@905924713 [redisClient=[addr=rediss://$IP_ADDR_C:6379], channel=[id: 0xab424c32, L:/$IP_ADDR_A:58362 - R:$IP_ADDR_C/$IP_ADDR_C:6379], currentCommand=null, usage=1]
timestamp=2022-07-06T12:01:39,070Z, level=TRACE, thread=redisson-netty-2-1, logger=o.r.client.handler.CommandEncoder, message=channel: [id: 0xab424c32, L:/$IP_ADDR_A:58362 - R:$IP_ADDR_C/$IP_ADDR_C:6379] message: *2
$6
EXISTS
$12
in-msg-prism

timestamp=2022-07-06T12:01:39,070Z, level=TRACE, thread=redisson-netty-2-6, logger=o.r.client.handler.CommandEncoder, message=channel: [id: 0xaa7da4bb, L:/$IP_ADDR_A:35790 - R:$IP_ADDR_B/$IP_ADDR_B:6379] message: *2
$6
EXISTS
$12
in-msg-prism

timestamp=2022-07-06T12:01:39,072Z, level=TRACE, thread=redisson-netty-2-1, logger=o.r.client.handler.CommandDecoder, message=reply: -MOVED 16324 cluster-001-0002-002.prt2ri.memorydb.us-east-1.amazonaws.com:6379
, channel: [id: 0xab424c32, L:/$IP_ADDR_A:58362 - R:$IP_ADDR_C/$IP_ADDR_C:6379], command: CommandData [promise=java.util.concurrent.CompletableFuture@3d602b64[Not completed, 1 dependents], command=(EXISTS), params=[in-msg-prism], codec=org.redisson.codec.MarshallingCodec]
timestamp=2022-07-06T12:01:39,072Z, level=DEBUG, thread=redisson-netty-2-1, logger=org.redisson.command.RedisExecutor, message=connection released for command (EXISTS) and params [in-msg-prism] from slot NodeSource [slot=null, addr=null, redisClient=null, redirect=null, entry=MasterSlaveEntry [masterEntry=[freeSubscribeConnectionsAmount=1, freeSubscribeConnectionsCounter=value:50:queue:0, freeConnectionsAmount=24, freeConnectionsCounter=value:64:queue:0, freezeReason=null, client=[addr=rediss://$IP_ADDR_D:6379], nodeType=MASTER, firstFail=0]]] using connection RedisConnection@905924713 [redisClient=[addr=rediss://$IP_ADDR_C:6379], channel=[id: 0xab424c32, L:/$IP_ADDR_A:58362 - R:$IP_ADDR_C/$IP_ADDR_C:6379], currentCommand=CommandData [promise=java.util.concurrent.CompletableFuture@3d602b64[Completed exceptionally: org.redisson.client.RedisMovedException], command=(EXISTS), params=[in-msg-prism], codec=org.redisson.codec.MarshallingCodec], usage=0]
timestamp=2022-07-06T12:01:39,076Z, level=TRACE, thread=redisson-netty-2-6, logger=o.r.client.handler.CommandDecoder, message=reply: :1
, channel: [id: 0xaa7da4bb, L:/$IP_ADDR_A:35790 - R:$IP_ADDR_B/$IP_ADDR_B:6379], command: CommandData [promise=java.util.concurrent.CompletableFuture@5ba176bd[Not completed, 1 dependents], command=(EXISTS), params=[in-msg-prism], codec=org.redisson.codec.MarshallingCodec]
timestamp=2022-07-06T12:01:39,076Z, level=DEBUG, thread=redisson-netty-2-6, logger=org.redisson.command.RedisExecutor, message=connection released for command (EXISTS) and params [in-msg-prism] from slot NodeSource [slot=null, addr=null, redisClient=null, redirect=null, entry=MasterSlaveEntry [masterEntry=[freeSubscribeConnectionsAmount=1, freeSubscribeConnectionsCounter=value:50:queue:0, freeConnectionsAmount=24, freeConnectionsCounter=value:64:queue:0, freezeReason=null, client=[addr=rediss://$IP_ADDR_E:6379], nodeType=MASTER, firstFail=0]]] using connection RedisConnection@1735807760 [redisClient=[addr=rediss://$IP_ADDR_B:6379], channel=[id: 0xaa7da4bb, L:/$IP_ADDR_A:35790 - R:$IP_ADDR_B/$IP_ADDR_B:6379], currentCommand=CommandData [promise=java.util.concurrent.CompletableFuture@5ba176bd[Completed normally], command=(EXISTS), params=[in-msg-prism], codec=org.redisson.codec.MarshallingCodec], usage=0]
timestamp=2022-07-06T12:01:39,078Z, level=ERROR, thread=redisson-netty-2-6, logger=s.empyrean.redis.keys.RedisKeyClient, message=[key=in-msg-prism] Failed to check if key exists
java.util.concurrent.CompletionException: org.redisson.client.RedisMovedException
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:332)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:347)
	at java.base/java.util.concurrent.CompletableFuture$BiRelay.tryFire(CompletableFuture.java:1498)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
	at org.redisson.command.RedisExecutor.handleReference(RedisExecutor.java:521)
	at org.redisson.command.RedisExecutor.handleSuccess(RedisExecutor.java:513)
	at org.redisson.command.RedisExecutor.handleResult(RedisExecutor.java:501)
	at org.redisson.command.RedisExecutor.checkAttemptPromise(RedisExecutor.java:478)
	at org.redisson.command.RedisExecutor.lambda$execute$3(RedisExecutor.java:172)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
	at org.redisson.client.handler.CommandDecoder.completeResponse(CommandDecoder.java:460)
	at org.redisson.client.handler.CommandDecoder.handleResult(CommandDecoder.java:454)
	at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:377)
	at org.redisson.client.handler.CommandDecoder.decodeCommand(CommandDecoder.java:198)
	at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:137)
	at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:113)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:510)
	<removed for brevity>
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.redisson.client.RedisMovedException: null
	at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:341)
	... 31 common frames omitted
```

**Comment by mrniko** [comment]
Thanks for the logs.

request is redirected to `cluster-001-0002-002.prt2ri.memorydb.us-east-1.amazonaws.com`. Which IP bounded to this host IP_ADDR_C or IP_ADDR_B ?

**Comment by tfn-smartthings** [comment]
I've recaptured the logs from when the service starts up until when we see this error. (I figured the additional full context might help.) You can see other operations like XADD work as expected before and during the issue with EXISTS. Additionally, at the top of this file there is a mapping key between the DNS names and the masked IPs, so hopefully it is more easy to understand. Thanks again!

New logs:
https://gist.githubusercontent.com/tfn-smartthings/d438dabaf78fbb289ec0c5e026125af6/raw/d340b3c41cbaebde226734419d2e73ccab0d16c3/RedissonLog.log

**Comment by tfn-smartthings** [comment]
I've been playing around with Redis/MemoryDB around this issue and have an observation to add:
I created a new MemoryDB cluster that matches the previous configuration mentioned before. Additionally I created a simple application that does nothing but call `RKeysAsync#countExistsAsync`. I made this call with a random key that has not been otherwise touched/created/etc... and I get the same error as above.

I'll try to dig into the Redisson code itself to see if I can find the problem, but putting this out there in case it rings a bell.
